src/ts/components/AppBar.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {Component} from 'react';
+import {Link} from 'react-router';
+import {AppBar} from 'material-ui';
+import Menu from 'components/Menu';
+
+// Scroll logic based on https://gist.github.com/Warry/4254579
+
+const barStyle = {
+  position: 'fixed',
+  transform: 'translate3d(0,0,0)',
+  MozTransform: 'translate3d(0,0,0)',
+  WebkitTransform: 'translate3d(0,0,0)'
+};
+
+const titleStyle = {
+  fontFamily: 'Junicode',
+  fontSize: 43,
+  color: '#AD000D',
+  padding: '3px 0 0 14px',
+  marginBottom: -3
+};
+
+export default class AppBarOwn extends Component<any, any> {
+
+  constructor(props) {
+    super(props);
+    this.state = {visible: false};
+    this.handleScroll = this.handleScroll.bind(this)
+  }
+
+  shouldComponentUpdate() {
+    return false;
+  }
+
+  componentDidMount() {
+    this.handleScroll();
+  }
+
+  componentWillUnmount() {
+    this.handleScroll = () => null;
+  }
+
+  handleScroll() {
+    if (this.state.visible && window.scrollY === 0) {
```

Comment:
Ok, I added `if (window && window.scrollY)` at the beginning, btw. the same situation was in [polyfill.js](https://github.com/michalmikolajczyk/viking-garage-frontend/blob/detail-view/src/js/pollyfills.js)